src/vmod_soap_request.c

```diff
@@ -30,33 +30,38 @@
 #include "vmod_soap_gzip.h"
 
 static ssize_t
-fill_pipeline(struct soap_req_http *req_http, struct http_conn *htc, ssize_t len)
+fill_pipeline(struct soap_req_http *req_http, struct http_conn *htc, body_part *pipeline, int bytes_read, int bytes_total)
 {
 	char *buf;
 	ssize_t l;
 	ssize_t i = 0;
+	int len;
 
 	CHECK_OBJ_NOTNULL(htc, HTTP_CONN_MAGIC);
-	assert(len > 0);
+	assert(bytes_total > bytes_read);
+	len = bytes_total - bytes_read;
 	l = 0;
 	if (htc->pipeline_b) {
 		l = htc->pipeline_e - htc->pipeline_b;
 		assert(l > 0);
-		if (l > len) {
-			return (l);
+		if (l >= len + bytes_read) {
```

Comment:
sorry I don't get the use-case for this if condition. Could you describe the case where it matches ? Thx in advance

Reply:
Added a comment above, to make it clear.